### PR TITLE
feat(sbb-train-formation): introduce new types and refactoring

### LINKED ISSUE #2681: [Change request]: changes for sbb-train-wagon (type "closed", sleeping car/couchette)
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/orgs/sbb-design-systems/projects/4) for an issue that matches the one I want to file, without success.

### Request type

Request for enhancement of a component

### It affects the following packages

Components

### Functionality

As discussed with SBB Mobile and sbb.ch we want to make a few slight changes on the display of the train formation.

- for sleeping cars and couchettes, the attribute icon should be placed inside the wagon frame
- the design of the closed wagon has to be changed: crossed-through frame

### Link to design proposal file

https://www.figma.com/design/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?m=auto&node-id=43654-271442&t=LMy9FQ5CJzbAblb5-1

### PR DESCRIPTION
**Pull request by jeripeierSBB**
Closes #2681

BREAKING CHANGE: The `hide-wagen-label` property of the `sbb-train-formation` was removed. Now it automatically doesn't show the label if no label is set on all the wagons. 
The i18n `i18nClosedCompartmentLabel()` method doesn't take `wagonNumber` as an argument anymore but is a constant now.
Additionally, there are some visual changes:
- `sbb-train-wagon`: The `ouccpancy` property doesn't default to `none` anymore but to `null`. Please replace the currently undefined occupancy property with the value `none`.
- `sbb-train-wagon`: Previously for the locomotive the label was not displayed, but now it would, as soon as there is one provided
- `sbb-train-formation`: The inline padding (left / right) was removed but can be set by CSS variable. See documentation.

### CODE REVIEW

File: src/elements/train/train-wagon/train-wagon.ts [hunk 1]
```diff
@@ -90,126 +101,137 @@ class SbbTrainWagonElement extends SbbNamedSlotListMixin<SbbIconElement, typeof
...
+        : nothing;
+
+    const wagonClassContent = html`<span class="sbb-screen-reader-only">
+        ${this.wagonClass === '1'
+          ? i18nClass['first'][this._language.current]
+          : i18nClass['second'][this._language.current]}
+      </span>
+      <span aria-hidden="true">${this.wagonClass}</span>`;
+
+    const mainIcon = choose(this.type, [
```
Comment (kyubisation) [inline src/elements/train/train-wagon/train-wagon.ts:152]: Nitpick; From my understanding, the only thing that changes is the icon name. Could this not be solved by just mapping the type to the icon name and setting the attribute accordingly?

File: src/elements/train/train/train.ts [hunk 2]
```diff
@@ -43,7 +43,8 @@ class SbbTrainElement extends SbbNamedSlotListMixin<
 
   /** General label for "driving direction". */
   @forceType()
-  @property({ attribute: 'direction-label' })
+  // Has to be reflected as it is used in core.scss as selector.
```
Comment (kyubisation) [inline src/elements/train/train/train.ts:46]: We don't generally write a comment about this. In my eyes the presence of the reflect automatically justifies this.
Or is there a reason you want to highlight it?

**Review by kyubisation — COMMENTED** [review]
Generally LGTM 👍 
Some questions

File: src/elements/train/train-formation/train-formation.scss [hunk 3]
```diff
@@ -68,34 +85,38 @@
...
 .sbb-train-formation__sector-sticky-wrapper {
-  background-color: var(--sbb-train-formation-background-color);
   position: sticky;
   inset: auto 0;
   padding: 0 var(--sbb-spacing-fixed-1x);
   margin: 0 var(--sbb-spacing-fixed-2x);
   overflow: hidden;
   white-space: nowrap;
+
+  &::before,
```
Comment (jeripeierSBB) [inline src/elements/train/train-formation/train-formation.scss:103]:
```suggestion
  // Lines beside the sector-label
  &::before,
```

File: src/elements/train/train-formation/train-formation.scss [hunk 4]
```diff
@@ -68,34 +85,38 @@
...
+  &::after {
+    content: '';
+    position: absolute;
+    inset-block: calc(50% - var(--sbb-train-formation-sector-line-width) / 2);
```
Comment (jeripeierSBB) [inline src/elements/train/train-formation/train-formation.scss:107]:
```suggestion
    // Exactly center the line by subtracting half of the border width
    inset-block: calc(50% - var(--sbb-train-formation-sector-line-width) / 2);
```

File: src/elements/train/train-wagon/train-wagon.ts [hunk 5]
```diff
@@ -90,126 +106,129 @@ class SbbTrainWagonElement extends SbbNamedSlotListMixin<SbbIconElement, typeof
...
-            this.label
-              ? html` <span class="sbb-screen-reader-only">
-                    ${`${i18nWagonLabelNumber[this._language.current]},`}&nbsp;
-                  </span>
-                  ${this.label}`
-              : nothing
-          }
-        </${unsafeStatic(TAG_NAME)}>
-      `;
+    const typeLabel = (): string => {
```
Comment (kyubisation) [inline src/elements/train/train-wagon/train-wagon.ts:114]: Nitpick; Could probably be extracted into a private method?

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Nice work! 😃

**Pull request merged**